devtools/cli/src/main/java/io/quarkus/cli/Create.java

```diff
@@ -0,0 +1,128 @@
+    String version = "1.0-SNAPSHOT";
+
+    @CommandLine.Option(names = { "-0",
+            "--skipExample" }, order = 4, description = "Generate without example code.")
+    boolean skipExample = false;
+
+    @CommandLine.ArgGroup()
+    TargetBuildTool targetBuildTool = new TargetBuildTool();
+
+    static class TargetBuildTool {
+        @CommandLine.Option(names = { "--maven" }, order = 5, description = "Create a Maven project. (default)")
+        boolean maven = false;
+
+        @CommandLine.Option(names = { "--gradle" }, order = 6, description = "Create a Gradle project.")
+        boolean gradle = false;
+
+        @CommandLine.Option(names = {
+                "--grade-kotlin-dsl" }, order = 7, description = "Create a Gradle Kotlin DSL project.")
+        boolean gradleKotlinDsl = false;
+    }
+
+    @CommandLine.ArgGroup()
+    TargetLanguage language = new TargetLanguage();
+
+    static class TargetLanguage {
+        @CommandLine.Option(names = {
+                "--java" }, order = 8, description = "Generate Java examples. (default)")
+        boolean java = false;
+
+        @CommandLine.Option(names = {
+                "--kotlin" }, order = 9, description = "Generate Kotlin examples.")
+        boolean kotlin = false;
+
+        @CommandLine.Option(names = {
+                "--scala" }, order = 10, description = "Generate Scala examples.")
+        boolean scala = false;
+    }
+
+    @CommandLine.Parameters(arity = "0..1", paramLabel = "EXTENSION", description = "extension to add to project")
+    Set<String> extensions;
+
+    @CommandLine.ParentCommand
+    QuarkusCli parent;
+
+    @CommandLine.Spec
+    CommandLine.Model.CommandSpec spec;
+
+    @Override
+    public Integer call() throws Exception {
+        try {
+            File projectDirectory = new File(System.getProperty("user.dir"));
+
+            File projectRoot = new File(projectDirectory, artifactId);
+            if (projectRoot.exists()) {
+                spec.commandLine().getErr().println("Unable to create the project, " +
+                        "the directory " + projectRoot.getAbsolutePath() + " already exists");
+                return ExitCodes.FAILURE;
+            }
+
+            SourceType sourceType = SourceType.JAVA;
+            if (targetBuildTool.gradleKotlinDsl)
+                sourceType = SourceType.KOTLIN;
+            else if (language.scala)
+                sourceType = SourceType.SCALA;
+            else if (language.kotlin)
+                sourceType = SourceType.KOTLIN;
+            else if (extensions != null && !extensions.isEmpty())
+                sourceType = CreateProject.determineSourceType(extensions);
+
+            BuildTool buildTool = BuildTool.MAVEN;
+            if (targetBuildTool.gradle)
+                buildTool = BuildTool.GRADLE;
+            else if (targetBuildTool.gradleKotlinDsl)
+                buildTool = BuildTool.GRADLE_KOTLIN_DSL;
```

Comment:
Looks like brackets are no longer a thing in 2020 :)

Reply:
Seriously?

Reply:
Well, yeah :) IMHO it looks better with brackets :)

Reply:
It's all fun and games until someone adds a new line and wonders why it doesn't work as it should :wink:

Reply:
I just don't agree. :) We're not coding in 'vi' or 'ed' anymore.